components/sysutils/scfdot/files/scfdotctl

```diff
@@ -0,0 +1,176 @@
+#!/bin/sh
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL)". You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2017 Andreas Grueninger, Grueninger GmbH, (grueni). All rights reserved.
+#
+
+# check the requirements
+if [ ! -f /usr/bin/dot ] || [ ! -f /usr/bin/gs ]; then
+echo "-> Ups! We need graphviz and ghostscript, please install the packages with:"
+echo "pkg install print/filter/ghostscript image/graphviz"
+exit
+fi
+
+# Build scfdot, invoke it to generate a graph description, and run dot to
+# render it to PostScript.  The resulting file will be <hostname>.ps .
+
+# Options to pass to scfdot.  This limits the graph to 300" by 42", includes
+# legend.ps as the legend (built below), and consolidates inetd services into
+# a single node.  See the comment at the top of scfdot.c for other options.
+
+SCFDOTOPTSH=-s\ 300,42\ -l\ legend.ps\ -x\ consolidate_inetd_svcs
+SCFDOTOPTSL='-L'
+
+# Margin, in inches, to include above and below the legend.
+# LEGEND_MARGIN=3
+
+# Options to pass to dot when rendering the graph.  Consider increasing
+# mclimit, which dictates how long dot spends optimizing node placement.  It
+# defaults to 1.0; 100 should produce good output, but it may take a long
+# time.
+DOTOPTS=-Gmclimit=100
+
+HOSTNAME=`hostname`
+
+cat << EOF > /tmp/setpage.awk
```

Comment:
Should we use mktemp or at least paid to generate unique file names for temporary files?

Reply:
s/paid/pid/